routing/index_router.cpp

```diff
@@ -302,62 +314,52 @@ IRouter::ResultCode IndexRouter::ProcessLeaps(vector<Segment> const & input,
 
   WorldGraph & worldGraph = starter.GetGraph();
   WorldGraph::Mode const worldRouteMode = worldGraph.GetMode();
-  worldGraph.SetMode(WorldGraph::Mode::SingleMwm);
+  worldGraph.SetMode(WorldGraph::Mode::NoLeaps);
 
   for (size_t i = 0; i < input.size(); ++i)
   {
     Segment const & current = input[i];
-    if (worldRouteMode == WorldGraph::Mode::LeapsIfPossible && !starter.IsLeap(current.GetMwmId()))
+
+    if ((worldRouteMode == WorldGraph::Mode::LeapsOnly && IndexGraphStarter::IsFakeSegment(current))
+      || (worldRouteMode != WorldGraph::Mode::LeapsOnly && !starter.IsLeap(current.GetMwmId())))
     {
       output.push_back(current);
       continue;
     }
 
+    // Clear previous loaded graphs to not spend too much memory at one time.
+    worldGraph.ClearIndexGraphs();
```

Comment:
It's for cross mwm routing case. After a route within one mwm (or from start to exits/from enter to finish) was created one or more IndexGraph can be loaded. This line clear unordered_map<NumMwmId, unique_ptr<IndexGraph>> IndexGraphLoaderImpl::m_graphs data before creating a leap route within one mwm.